index.bs

```diff
@@ -4619,18 +4677,27 @@ script=].
 <div algorithm="remote end steps for script.addPreloadScript">
 The [=remote end steps=] given |session| and |command parameters| are:
 
-1. Let |expression| be the <code>expression</code> field of |command
+1. Let |function declaration| be the <code>function declaration</code> field of |command
    parameters|.
 
+1. Let |arguments| be the <code>arguments</code> field of |command
+   parameters| if present, or an empty [=/list=] otherwise.
+
 1. Let |script| be the string representation of a [[!RFC4122|UUID]].
 
 1. Let |sandbox| be the value of the "<code>sandbox</code>" field in |command
    parameters|, if present, or null otherwise.
 
+1. Issue: We could parse the script at this point so we can response with an
+   error if the script doesn't evaluate as a function. However it's not clear
+   which realm would be used for that evaluation (we don't want to actually add
+   the function to any globals).
```

Comment:
Hi @sadym-chromium. Could you maybe help here and tell us if CDP parses the bootstrap script when it gets added, or is the parsing happening once it gets injected and an error is returned? If it parses / validates it directly it would be good to know how this is done.

Reply:
The script is parsed at the moment it is executed.